Mac is not happy... `/System/Volumes/Data/build/ci_checks/alice_ci_0/o2/sw/SOURCES/O2/6753/0/Analysis/Tasks/PWGLF/spectraTPCPiKaPr.cxx:52:9: error: value of type 'o2::framework::expressions::Node' is not contextually convertible to 'bool'
    if (abs(nsigma) > cfgNSigmaCut) {`

--- a/Analysis/Tutorials/src/conditionalExpressions.cxx
+++ b/Analysis/Tutorials/src/conditionalExpressions.cxx
@@ -20,7 +20,7 @@ using namespace o2::framework::expressions;
 
 struct ConditionalExpressions {
   Configurable<bool> useFlags{"useFlags", false, "Switch to enable using track flags for selection"};
-  Filter trackFilter = nabs(aod::track::eta) < 0.9f && aod::track::pt > 0.5f && ifnode(useFlags == true, (aod::track::flags & static_cast<uint32_t>(o2::aod::track::ITSrefit)) != 0u, true);
+  Filter trackFilter = nabs(aod::track::eta) < 0.9f && aod::track::pt > 0.5f && ifnode(useFlags.node() == true, (aod::track::flags & static_cast<uint32_t>(o2::aod::track::ITSrefit)) != 0u, true);
   OutputObj<TH2F> etapt{TH2F("etapt", ";#eta;#p_{T}", 201, -2.1, 2.1, 601, 0, 60.1)};
   void process(aod::Collision const&, soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra>> const& tracks)
   {

--- a/Framework/Core/include/Framework/Configurable.h
+++ b/Framework/Core/include/Framework/Configurable.h
@@ -15,6 +15,11 @@
 #include <vector>
 namespace o2::framework
 {
+namespace expressions
+{
+struct PlaceholderNode;
+}
+
 template <typename T, ConfigParamKind K>
 struct ConfigurableBase {
   ConfigurableBase(std::string const& name, T&& defaultValue, std::string const& help)
@@ -67,6 +72,10 @@ struct Configurable : IP {
   Configurable(std::string const& name, T&& defaultValue, std::string const& help)
     : IP{name, std::forward<T>(defaultValue), help}
   {
+  }
+  auto node()
+  {
+    return expressions::PlaceholderNode{*this};
   }
 };
 

--- a/Framework/Core/include/Framework/Expressions.h
+++ b/Framework/Core/include/Framework/Expressions.h
@@ -128,7 +128,7 @@ struct OpNode {
 /// A placeholder node for simple type configurable
 struct PlaceholderNode : LiteralNode {
   template <typename T>
-  PlaceholderNode(Configurable<T> v) : LiteralNode{v.value}, name{v.name}
+  PlaceholderNode(Configurable<T> const& v) : LiteralNode{v.value}, name{v.name}
   {
     if constexpr (variant_trait_v<typename std::decay<T>::type> != VariantType::Unknown) {
       retrieve = [](InitContext& context, std::string const& name) { return LiteralNode::var_t{context.options().get<T>(name.c_str())}; };
@@ -198,65 +198,55 @@ struct Node {
 
 /// overloaded operators to build the tree from an expression
 
-#define BINARY_OP_NODES(_operator_, _operation_)                                          \
-  template <typename T>                                                                   \
-  inline Node operator _operator_(Node left, T right)                                     \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, std::move(left), LiteralNode{right}};       \
-  }                                                                                       \
-  template <typename T>                                                                   \
-  inline Node operator _operator_(T left, Node right)                                     \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, LiteralNode{left}, std::move(right)};       \
-  }                                                                                       \
-  template <typename T>                                                                   \
-  inline Node operator _operator_(Node left, Configurable<T> right)                       \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, std::move(left), PlaceholderNode{right}};   \
-  }                                                                                       \
-  template <typename T>                                                                   \
-  inline Node operator _operator_(Configurable<T> left, Node right)                       \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, PlaceholderNode{left}, std::move(right)};   \
-  }                                                                                       \
-  inline Node operator _operator_(Node left, Node right)                                  \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, std::move(left), std::move(right)};         \
-  }                                                                                       \
-  inline Node operator _operator_(BindingNode left, BindingNode right)                    \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, left, right};                               \
-  }                                                                                       \
-  template <>                                                                             \
-  inline Node operator _operator_(BindingNode left, Node right)                           \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, left, std::move(right)};                    \
-  }                                                                                       \
-  template <>                                                                             \
-  inline Node operator _operator_(Node left, BindingNode right)                           \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, std::move(left), right};                    \
-  }                                                                                       \
-                                                                                          \
-  template <typename T>                                                                   \
-  inline Node operator _operator_(Configurable<T> left, BindingNode right)                \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, PlaceholderNode{left}, right};              \
-  }                                                                                       \
-  template <typename T>                                                                   \
-  inline Node operator _operator_(BindingNode left, Configurable<T> right)                \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, left, PlaceholderNode{right}};              \
-  }                                                                                       \
-  template <typename T, typename L>                                                       \
-  inline Node operator _operator_(Configurable<T> left, L right)                          \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, PlaceholderNode{left}, LiteralNode{right}}; \
-  }                                                                                       \
-  template <typename T, typename L>                                                       \
-  inline Node operator _operator_(L left, Configurable<T> right)                          \
-  {                                                                                       \
-    return Node{OpNode{BasicOp::_operation_}, left, PlaceholderNode{right}};              \
+#define BINARY_OP_NODES(_operator_, _operation_)                                        \
+  template <typename T>                                                                 \
+  inline Node operator _operator_(Node left, T right)                                   \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, std::move(left), LiteralNode{right}};     \
+  }                                                                                     \
+  template <typename T>                                                                 \
+  inline Node operator _operator_(T left, Node right)                                   \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, LiteralNode{left}, std::move(right)};     \
+  }                                                                                     \
+  template <typename T>                                                                 \
+  inline Node operator _operator_(Node left, Configurable<T> right)                     \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, std::move(left), PlaceholderNode{right}}; \
+  }                                                                                     \
+  template <typename T>                                                                 \
+  inline Node operator _operator_(Configurable<T> left, Node right)                     \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, PlaceholderNode{left}, std::move(right)}; \
+  }                                                                                     \
+  inline Node operator _operator_(Node left, Node right)                                \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, std::move(left), std::move(right)};       \
+  }                                                                                     \
+  inline Node operator _operator_(BindingNode left, BindingNode right)                  \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, left, right};                             \
+  }                                                                                     \
+  template <>                                                                           \
+  inline Node operator _operator_(BindingNode left, Node right)                         \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, left, std::move(right)};                  \
+  }                                                                                     \
+  template <>                                                                           \
+  inline Node operator _operator_(Node left, BindingNode right)                         \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, std::move(left), right};                  \
+  }                                                                                     \
+                                                                                        \
+  template <typename T>                                                                 \
+  inline Node operator _operator_(Configurable<T> left, BindingNode right)              \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, PlaceholderNode{left}, right};            \
+  }                                                                                     \
+  template <typename T>                                                                 \
+  inline Node operator _operator_(BindingNode left, Configurable<T> right)              \
+  {                                                                                     \
+    return Node{OpNode{BasicOp::_operation_}, left, PlaceholderNode{right}};            \
   }
 
 BINARY_OP_NODES(&, BitwiseAnd);

